src/controllers/api.controller.ts

```diff
@@ -139,3 +219,42 @@ export const claimsFeed = async (req: Request, res: Response, next: NextFunction
     passToExpressErrorHandler(err, next);
   }
 };
+
+async function populateImagesSignedUrls(imgs: Image[]) {
+  for (let i = 0; i < imgs.length; i++) {
+    imgs[i].url = await getS3SignedUrl(imgs[i].url);
+  }
+}
+
+export async function populateReportImagesSignedUrls(report: Report) {
+  if (report.edge) await populateEdgeImagesSignedUrls(report.edge);
+
+  report.claim.image = await getS3SignedUrlIfExisted(report.claim.image);
+
+  for (let i = 0; i < report.claim.relatedNodes.length; i++) {
+    await populateImagesSignedUrls(report.claim.images);
+  }
+
+  await populateImagesSignedUrls(report.claim.images);
+
+  for (let i = 0; i < report.claim.relatedNodes.length; i++) {
+    await populateNodeImagesSignedUrls(report.claim.relatedNodes[i]);
+  }
+  for (let i = 0; i < report.validations.length; i++) {
+    await populateNodeImagesSignedUrls(report.validations[i]);
+  }
+  for (let i = 0; i < report.attestations.length; i++) {
+    await populateNodeImagesSignedUrls(report.attestations[i]);
+  }
+}
+
+async function populateEdgeImagesSignedUrls(edge: Edge & { startNode: Node; endNode: Node }) {
+  edge.thumbnail = await getS3SignedUrlIfExisted(edge.thumbnail);
```

Comment:
we shoudl not be splat s3 everywhere

not all images are stored in s3

its possible image is just a url to somewhere not our storage

Reply:
I didn't know that was the case!

The simplest fix I can think of rn is to expose S3's bucket to be publicly available to read from and store the full url in the DB. Otherwise, we'll need to add another field in the database to store the source of the image, and that will require a DB migration.

Let me know which option should I go on with to start working on

Reply:
OR, I can check if the image field is a full url or base64 then I don't have to sign it. I think the only case that it won't be a url or base64 is it is an S3 key! This should be the easier solution, I think

Reply:
But that also means that the user can send the image URL directly besides uploading an image to our S3 bucket. The UI doesn't have this option, and the backend too doesn't have this option, it wasn't indicated anywhere

Reply:
hey @omareloui you NEVER have to sign it please rip out all signing stuff this is totally unnecessary complexity - leave the field in the db OPTIONAL for possible future use we DO NOT CARE about any signing at all rip all that out completely it will just break things later

Reply:
I'll remove S3's signature assuming that the bucket will be publicly available to read from and instead of storing the key I'll store the full URL